lua/dap-go.lua

```diff
@@ -76,6 +76,8 @@ local function setup_adapter()
       :start()
 
     vim.defer_fn(function()
+      -- I have no idea why this is required, I suspect the firewall or something else?
+      os.execute("sleep " .. tonumber(1))
```

Comment:
This is the definition of empirical. I suspect there is a timing issue here. I don't know what would be the best way to fix that.

Reply:
@jhchabran do you have the error message of this one?

You are adding a sleep instruction inside the defer of the callback. That should not be necessary as the defer is scheduled to run after the 100 ms timeout. 

I suspect that your system is taking longer to start the server. Could you use a value higher and see if the behaviour is fixed. If it works I believe that a better approach would be make the timeout configurable.

Reply:
> You are adding a sleep instruction inside the defer of the callback. That should not be necessary as the defer is scheduled to run after the 100 ms timeout.

Oh, I totally missed the later argument with the `100` ms timeout, that's much much better. Sloppy Sunday hacking 😆 

I just tried it again, but strangely I'm not able to reproduce it, even when lowering the timeout to 10ms. 

The error I was getting was the `ERRCONNREFUSED` from failing to reach the `dlv` server.